notification-server/src/processor.rs

```diff
@@ -0,0 +1,78 @@
+use concordium_rust_sdk::{
+    cis2,
+    cis2::Event,
+    types,
+    types::{
+        AccountTransactionEffects, BlockItemSummaryDetails::AccountTransaction,
+        ContractTraceElement,
+    },
+};
+use concordium_rust_sdk::base::contracts_common::AccountAddress;
+use concordium_rust_sdk::types::Address::Account;
+use futures::{Stream, StreamExt};
+
+fn get_cis2_events_addresses(effects: &AccountTransactionEffects) -> Option<Vec<AccountAddress>> {
+    match &effects {
+        AccountTransactionEffects::ContractUpdateIssued { effects } => Some(
+            effects
+                .iter()
+                .flat_map(|effect| match effect {
+                    ContractTraceElement::Updated { data } => data
+                        .events
+                        .iter()
+                        .map(|event| match cis2::Event::try_from(event) {
+                            Ok(Event::Transfer { to, .. }) => Some(to),
+                            Ok(Event::Mint { owner, .. }) => Some(owner),
+                            _ => None,
+                        })
+                        .filter_map(|addr| match addr {
+                            Some(Account(addr)) => Some(addr),
+                            _ => None,
+                        })
+                        .collect(),
+                    _ => vec![],
+                })
+                .collect(),
+        ),
+        _ => None,
+    }
+}
+
+fn is_notification_emitting_transaction_effect(effects: &AccountTransactionEffects) -> bool {
+    matches!(
+        effects,
+        AccountTransactionEffects::AccountTransfer { .. }
+            | AccountTransactionEffects::AccountTransferWithMemo { .. }
+            | AccountTransactionEffects::TransferredWithSchedule { .. }
+            | AccountTransactionEffects::TransferredWithScheduleAndMemo { .. }
```

Comment:
One additional option that CCD can be sent to an account would be that a smart contract sends CCD to an account address. This coudl be added later but wanted to mention that this case has not been handled yet.

Reply:
isent that covered by:

Event::Transfer?

Reply:
Not the `cis2::Event::Transfer` but `ContractTraceElement::Transferred { to, from, amount }`